src/target/riscv/startup.tcl

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
```

Comment:
Just a comment - no change requested:

I have not reviewed the changes in this file in detail - I trust you on those.

The new type of tests that @en-sc has proposed [here](https://review.openocd.org/c/openocd/+/8507) would be very good for testing changes like these. Perhaps you are using such tests already internally?

Reply:
> The new type of tests that @en-sc has proposed [here](https://review.openocd.org/c/openocd/+/8507) would be very good for testing changes like these. Perhaps you are using such tests already internally?

1. These types of tests would be useful, though integrating it to that infrastructure will be difficult, since these specific tests require a specific response from the target (dummy adapter just  guarantees that some results are returned, not a valid one). 
2. This commit is tested internally against using internal testsuite that I'm still in the process of opensourcing :(. Those test just generate MMU pages on the fly, flip the switch for translation mode and try to do memory reads/writes  comparing results with the expected one.

Reply:
Answered from the wrong account, sorry for spam.

> The new type of tests that @en-sc has proposed [here](https://review.openocd.org/c/openocd/+/8507) would be very good for testing changes like these. Perhaps you are using such tests already internally?

1. These types of tests would be useful, though integrating it to that infrastructure will be difficult, since these specific tests require a specific response from the target (dummy adapter just  guarantees that some results are returned, not a valid one). 
2. This commit is tested internally against using internal testsuite that I'm still in the process of opensourcing :(. Those test just generate MMU pages on the fly, flip the switch for translation mode and try to do memory reads/writes  comparing results with the expected one.